### PR TITLE
Clean up the repo and remove Xodr/XoscSchemaChecker

### LINKED ISSUE #7: Remove capitals from file names and use snake_case instead
**Is your feature request related to a problem? Please describe.**

Capitals may lead to problems in Windows/Linux mixed environments. And file names should look consistent.

**Describe the solution you'd like**

Use snake_case and only lower case.

### LINKED ISSUE #8: Remove XSD checker (OpenDRIVE & OpenSCENARIO XML) from framework
**Is your feature request related to a problem? Please describe.**

The framework shall be independent from specific standards. The existing schema checks will be replaced by the one which are developed within the Checker Libraries
* https://github.com/asam-ev/qc-opendrive
* https://github.com/asam-ev/qc-openscenarioxml

**Describe the solution you'd like**

Remove currently delivered XSD Checker Bundles from framework, documentation and tests.

### LINKED ISSUE #44: Add .clang-format file for code formatting
**Is your feature request related to a problem? Please describe.**

We need to add a `.clang-format` file to ensure consistency of C++ code style.

Note that, for Python, we will use [black](https://github.com/psf/black).

**Describe the solution you'd like**

Add a `.clang-format` file and reformat the repository.

### PR DESCRIPTION
**Pull request by hoangtungdinh**
**Description**

This PR includes changes related to the first development phase of the ASAM Quality Checker project. The changes in this PR are ready for CCB's review.

The main changes are listed as a list of smaller PRs below. For the detailed code change per feature, please check in the individual PR.

**Main changes**
1. #46
2. #47
3. #48
4. #49

**How was the PR tested?**

Tests are documented in each PR listed above.

**Notes**

This PR will close the following issues
* Close #7 
* Close #8 
* Close #44

### CODE REVIEW

**Review by andreaskern74 — APPROVED** [review]

File: src/common/src/result_format/c_checker.cpp [hunk 1]
```diff
@@ -257,41 +254,41 @@ void cChecker::ClearParams()
 }
 
 /*
```
Comment (MatteoRagni) [inline src/common/src/result_format/c_checker.cpp:256]:
```suggestion
/**
```
To enable automatic docs, I suspect this is required. To be repeated in the rest of the file.

File: src/common/src/result_format/c_issue.cpp [hunk 2]
```diff
@@ -267,16 +268,16 @@ std::string PrintIssueLevel(const eIssueLevel level)
...
+        break;
+    case WARNING_LVL:
+        retval = "Warning";
+        break;
+    default:
+        retval = "Error";
     }
-return retval;
+    return retval;
 }
```
Comment (MatteoRagni) [inline src/common/src/result_format/c_issue.cpp:283]:
```
std::string PrintIssueLevel(const eIssueLevel level)
{
    switch (level)
    {
    case INFO_LVL:
        return "Information";
    case WARNING_LVL:
        return "Warning";
    default:
        return "Error";
    }
}
```

I don't know why, Github does not allow me to insert a suggestion

**Review by MatteoRagni — COMMENTED** [review]
I was not able to complete everything in review (I stopped after ~80 files), but I'm quite confident it is good. I've put some comments that reflect personal preferences in coding. Feel free to ingore them, in particoular if they are on automatic generated code from Xerces.

I'm not approving directly because I'm not able to test the PR code on my machine. If @andreaskern74  approves it, I will agree, also because he has a very deep understanding of the codebase that I'm lacking.

**Comment by MatteoRagni** [comment]
> I was not able to complete everything in review (I stopped after ~80 files), but I'm quite confident it is good. I've put some comments that reflect personal preferences in coding. Feel free to ingore them, in particoular if they are on automatic generated code from Xerces.
> 
> I'm not approving directly because I'm not able to test the PR code on my machine. If @andreaskern74 approves it, I will agree, also because he has a very deep understanding of the codebase that I'm lacking.

EDIT: @andreaskern74 approved it while I was rewieving :smiley:

**Comment by hoangtungdinh** [comment]
Thanks a lot for the review @MatteoRagni . Please feel free to review the PRs, even if they are already closed. We also discussed in the CCB meeting today that reviews on closed PRs are welcome. They will be addressed in the future PRs.

If it's convenient for you, you could also do the code review in every single "develop" PR in the future, so that it's easier to follow the code changes. The only logical change we made to the code is the removal of xsd checker #49. The rest is code from the original codebase and your feedback is very helpful to improve it further.

**Pull request merged**